core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableMergeJoin.java

```diff
@@ -194,6 +250,81 @@ public static boolean isMergeJoinSupported(JoinRelType joinType) {
     return DeriveMode.BOTH;
   }
 
+  private List<Integer> immutableIntListToList(ImmutableIntList intList, int offset) {
```

Comment:
I will probably add more comments to these new methods to explain what they are doing.

Reply:
Can you rebase on master? I just updated this file.
Use `ImmutableIntList.incr` instead.